.circleci/config.yml

```diff
@@ -297,14 +297,15 @@ defaults:
   - base_osx: &base_osx
       macos:
         xcode: "13.2.0"
+      resource_class: macos.x86.medium.gen2
       environment:
         TERM: xterm
         MAKEFLAGS: -j5
 
   - base_osx_large: &base_osx_large
       macos:
         xcode: "13.2.0"
-      resource_class: large
+      resource_class: macos.m1.large.gen1
```

Comment:
Got it

Reply:
Thanks. But I guess the PR will require a bit more work that it seemed at first.

As far as I know, we never build for an ARM-based architecture, so it will require more changes in our CI scripts to make everything work properly. For instance, see the errors here: https://app.circleci.com/pipelines/github/ethereum/solidity/29456/workflows/3a5b4dae-a538-43b0-807e-ee4d5c48b77b/jobs/1308373

Reply:
Not sure. We will need to fix this eventually anyway. And we are planning to work on that eventually, but it is not in our priority roadmap right now.

But if you want to continue work on that, I would recommend you to look the errors in the CI and see how they can be fixed. For instance, the first one seems to be related with this line: https://github.com/ethereum/solidity/blob/develop/.circleci/osx_install_dependencies.sh#L55 (I guess python3 is already the default in `m1`, based on the list of installed software in the `m1` image). So we need to either remove this line from that script, if this is also true for the `macos.x86.medium.gen2` image, or we need to adapt the script to work on both images.

The second error seems to be related to `bash` [not being installed or the default shell](https://circle-macos-docs.s3.amazonaws.com/image-manifest/v11441/manifest.txt) in their `m1` image or maybe it is installed but it does not have `shopt` built-in command? Not really sure, I'm not a mac user.

Reply:
I have removed brew unlink python in the osx_install_dependencies.sh file. Also from this post: https://github.com/mig1984/bashible/issues/10, they seem to be having the same issue, and ```replacing shop-s inherit_errexit``` with ```shopt -s inherit_errexit 2>/dev/null || true``` seemed to have worked. Also I should probably re-add the base_osx_large part for the config.yml file. What do you think @r0qs?

Reply:
Let's not get into M1 here. We do have an issue for that but it's a bigger task while this one was supposed to be only a simple tweak to switch to the new machine sizes. And in any cases, it's not a replacement for `large` for us because that machine type is used by `b_osx`. `b_osx` is the job that produces the release macOS binary and has to be on x86. When we add M1, it will be in addition it, not as a replacement for it.